### PR TITLE
Fix crash in scenario selection menu

### LINKED ISSUE #46392: Crash in scenario selection menu on smaller terminal height
# Describe bug 

Game crashes when going into scenario select menu 

# Steps To Reproduce

Create new character, go to next screen from pool point selection. OR
Create new random character go back to scenario menu 

# Expected behavior

It shouldn't crash but display posibble scenarios 

# Screenshots

# Versions and configuration

# Additional context

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: None```

#### Purpose of change

Fixes #46392

### CODE REVIEW

**Comment by actual-nh** [comment]
For applicable OSes, is SIGWINCH being dealt with (might be internally in the SDL/ncurses libraries)?

File: src/newcharacter.cpp [hunk 1]
```diff
@@ -2107,8 +2107,9 @@ tab_direction set_scenario( avatar &u, points_left &points,
         const int w_location_h = 3;
         const int w_vehicle_h = 3;
         const int w_initial_date_h = 6;
-        const int w_flags_h = iContentHeight -
-                              ( w_sorting_h + w_profession_h + w_location_h + w_vehicle_h + w_initial_date_h );
+        const int w_flags_h = clamp<int>( iContentHeight -
+                                          ( w_sorting_h + w_profession_h + w_location_h + w_vehicle_h + w_initial_date_h ),
+                                          0, iContentHeight );
```
Comment (ZhilkinSerg) [inline src/newcharacter.cpp:2112]:
```suggestion
                                          0, iContentHeight - 1 );
```

File: src/newcharacter.cpp [hunk 2]
```diff
@@ -2107,8 +2107,9 @@ tab_direction set_scenario( avatar &u, points_left &points,
         const int w_location_h = 3;
         const int w_vehicle_h = 3;
         const int w_initial_date_h = 6;
-        const int w_flags_h = iContentHeight -
-                              ( w_sorting_h + w_profession_h + w_location_h + w_vehicle_h + w_initial_date_h );
+        const int w_flags_h = clamp<int>( iContentHeight -
+                                          ( w_sorting_h + w_profession_h + w_location_h + w_vehicle_h + w_initial_date_h ),
+                                          0, iContentHeight - 1 );
```
Comment (ZhilkinSerg) [inline src/newcharacter.cpp:2112]:
```suggestion
        const int w_flags_h = std::max( iContentHeight -
                                          ( w_sorting_h + w_profession_h + w_location_h + w_vehicle_h + w_initial_date_h ),
                                          0 );
```

Comment (ZhilkinSerg) [inline src/newcharacter.cpp:2112]:
```suggestion
        const int w_flags_h = clamp<int>( 0,
                                          iContentHeight -
                                          ( w_sorting_h + w_profession_h + w_location_h + w_vehicle_h + w_initial_date_h ),
                                          iContentHeight );
```

**Comment by ZhilkinSerg** [comment]
> 
> 
> For applicable OSes, is SIGWINCH being dealt with (might be internally in the SDL/ncurses libraries)?

#37894

**Pull request merged**